libraries/AP_Math/SCurve.cpp

```diff
@@ -1119,12 +1119,9 @@ void SCurve::set_kinematic_limits(const Vector3p &origin, const Vector3p &destin
                                   float speed_xy, float speed_up, float speed_down,
                                   float accel_xy, float accel_z)
 {
-    // ensure arguments are positive
-    speed_xy = fabsf(speed_xy);
-    speed_up = fabsf(speed_up);
-    speed_down = fabsf(speed_down);
-    accel_xy = fabsf(accel_xy);
-    accel_z = fabsf(accel_z);
+    if (is_negative(speed_xy) ||is_negative(speed_up) ||is_negative(speed_down) ||is_negative(accel_xy) ||is_negative(accel_z)) {
+        INTERNAL_ERROR(AP_InternalError::error_t::flow_of_control);
+    }
```

Comment:
It's at least theoretically possible now that negative values could creep in so we should test to see what will happen to the vehicle in this case.  I think this can happen if the user sets one of the speeds (perhaps speed down) to a negative number and disables arming checks.  Of course they shouldn't do that but they could so we should check what happens.  TBH, I'm not actually sure that we should remove these protections, it's very little code and we already get enough pings from overzealous people fuzz testing our parameters and then reporting bad things happen

Reply:
Hi @lthall, to be clear, I think maybe we should leave the fabsf() in.